### ISSUE TITLE
value error  while loading for more than 9999 values in dict 

### ISSUE THREAD

**Issue by viyatgandhi** [selik/xport#4]
```
Traceback (most recent call last):
  File "main.py", line 124, in <module>
    rows =  xport.load(f)
  File "build/bdist.linux-x86_64/egg/xport.py", line 380, in load
  File "build/bdist.linux-x86_64/egg/xport.py", line 164, in __iter__
  File "build/bdist.linux-x86_64/egg/xport.py", line 339, in _read_observations
ValueError: Incomplete record, ''
```

whenever i try to load a xpt file with more than 9999 values it gives me incomplete record error.

I am using below method for generating xpt file, but I have 15 columns or so 

```
mapping = {'numbers': [1, 3.14, 42],
           'text': ['life', 'universe', 'everything']}

# as a mapping of labels to columns
with open('answers.xpt', 'wb') as f:
    dump(f, mapping, mode='columns')
```

but while loading i am using below method

`rows =  xport.load(f)`

everything works fine if I loop till 9999 for inserting data,
but for 10000 records it gives me above error

Also, I am working on debugging the code and identify the problem.  Let me know if you can find something.

**Comment by selik** [comment]
@viyatgandhi Strange. I'll check it out.

**Comment by viyatgandhi** [comment]
@selik thanks. Let me know if you need any info. Same happens for exporting csv too if xpt has >=10000 lines of data.  
`python -m xport example.xpt > example.csv`

**Comment by selik** [comment]
I think I fixed it (d0b7bd2649546f23732c98be7f86f6aeb4679be5).

The XPORT spec allows for multiple tables in the same file, but I've never found an example of that, so I don't really know how it works. Because of that, I've got some paranoid end-of-file padding checks. The XPORT file must be padded to make sure the total size is a multiple of 80 bytes (because SAS thinks you might be viewing the raw file on an 80-characters-per-line terminal, I guess). If the size of an observation record multiplied by the number of observations was a multiple of 80, then the last block read would be empty and my check to make sure it was all padding would be false. I fixed it to pass the check if the last block is empty *or* all padding.
